setup.py

```diff
@@ -4,9 +4,38 @@
 import codecs
 import os
 import re
+import sys
 
 from setuptools import setup
 
+CURRENT_PYTHON = sys.version_info[:2]
+REQUIRED_PYTHON = (3, 6)
+
+# This check and everything above must remain compatible with Python 2.7.
+if CURRENT_PYTHON < REQUIRED_PYTHON:
+    sys.stderr.write(
+        """
+==========================
+Unsupported Python version
+==========================
+This version of urllib3 requires Python {}.{}, but you're trying to
+install it on Python {}.{}.
+This may be because you are using a version of pip that doesn't
+understand the python_requires classifier. Make sure you
+have pip >= 9.0 and setuptools >= 24.2, then try again:
```

Comment:
Will the `setuptools >= 24.2` python_requires code path ever be hit? It will fail here first

Reply:
It will show up as `Requires-Python: >=3.6, <4` in the wheel METADATA file, which means clients with Python 3.5 or lower will download 1.x automatically

Reply:
right they see the wheel, so `setuptools` is not hit here